### PR TITLE
MRG, ENH: Add array-spacing plugin and reorganize deps

### PR DESCRIPTION
**Pull request by larsoner**
Playing around with CIs a bit:

1. Add [flake8-array-spacing](https://github.com/larsoner/flake8-array-spacing) plugin. It allows us to remove some `# noqa` lines, which is nice. Not 100% sure we want this, but I want to see if it works at least.
2. Remove `pytest`, `flake8`, and related things from the `mne` env and `requirements.txt`. Now these files only contain what is necessary to get 100% functionality out of MNE-Python.
3. Create `requirements_testing.txt` and `doc/requirements.txt` with the *additional* requirements for running tests and building docs, respectively.

This simplifies our CI code a bit, which is nice. I'm not expecting the CIs to come back happy on the first try, but folks should feel free to have a look and see if they're happy with the overall structure for when I can get the CIs to cooperate.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -151,7 +129,7 @@ jobs:
       conda env update --file $CONDA_ENV
       pip uninstall -yq mne
       pip install -ve .
-      pip install pytest pytest-cov pytest-timeout pytest-sugar pytest-xdist flake8 codecov
+      pip install -r requirements_testing.txt
```
Comment (drammock) [inline azure-pipelines.yml:132]: lost `codecov` on this line...  why not include it in `requirements_testing.txt`?  Looks like most other lines using `requirements_testing.txt` also need `codecov`

Comment (larsoner) [reply azure-pipelines.yml:132]: You only need `codecov` if you are pushing coverage, i.e., you are a CI provider. devs don't need it

**Comment by larsoner** [comment]
I went back and forth on the naming, too. But I figured having one under `doc/` was better than another root-level file. But we don't have a better place for the `_testing.txt` one. I guess I could rename `tools` to `testing`...?

**Comment by drammock** [comment]
> I figured having one under `doc/` was better than another root-level file.

A bloated root-level doesn't bother me, FWIW, as long as the filenames are well organized.  `requirements.txt`, `requirements_testing.txt`, and `requirements_docs.txt` seems fine to me.  I'd also change `server_environment.yml` to `environment_server.yml` or perhaps `environment_headless.yml`

**Comment by larsoner** [comment]
> I'd also change server_environment.yml to environment_server.yml or perhaps environment_headless.yml

Let's hold off on this one, I'm hoping that maybe we won't need it after https://github.com/pyvista/pyvista/pull/824 (?)

**Comment by agramfort** [comment]
thx @larsoner 

we'll deal with notebook test failure next week

**Pull request merged**